### PR TITLE
feat/attention component

### PR DESCRIPTION
**Pull request by HellraiserCW**
Toaster component 
- with max-height css animation,
- with new color palette

### CODE REVIEW

File: src/app/services/toaster.service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { ToasterMessage } from '../app.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ToasterService {
+  private isHidden$: WritableSignal<boolean> = signal(true);
+  isHidden: Signal<boolean> = computed(this.isHidden$);
+  private toasterSettings$: WritableSignal<ToasterMessage> = signal(initialToasterConfig);
+  toasterSettings: Signal<ToasterMessage> = computed(this.toasterSettings$);
```
Comment (GennadiyShpak) [inline src/app/services/toaster.service.ts:12]: format order of this propperty by our code style

File: src/app/shared/components/epm-toaster/epm-toaster.component.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<div class="toaster__wrapper">
+  <h4 class="toaster__title">{{ toasterMessage().title }}</h4>
+  <p class="toaster__description">
+    <span>{{ toasterMessage().description }}</span>
+    <a class="toaster__toaster-link" [routerLink]="toasterMessage().url">{{ toasterMessage().linkedText }}</a>
+    <span>.</span>
+  </p>
+  <button class="toaster__close-btn" type="button" (click)="onCloseClick()">
+    <img [ngSrc]="toasterIcons.Close" width="24" height="24" alt="close toaster" />
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-toaster/epm-toaster.component.html:9]:
```suggestion
    <img [ngSrc]="toasterIcons.Close" width="24" height="24" alt="close toaster icon" />
```

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  selector: 'epm-toaster',
+  standalone: true,
+  imports: [CommonModule, NgOptimizedImage, RouterLink],
+  templateUrl: './epm-toaster.component.html',
+  styleUrls: ['./epm-toaster.component.scss']
+})
+export class EpmToasterComponent implements OnInit {
+  @Input() title = '';
+
+  @HostBinding('class.hidden') animationClass = true;
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-toaster/epm-toaster.component.ts:19]: i think name animationClass don`t discribe its purpose

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  toasterMessage!: Signal<ToasterMessage>;
+
+  readonly routerNames = APP_ROUTER_NAME;
+  readonly toasterIcons = TOASTER_ICONS;
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-toaster/epm-toaster.component.ts:24]: where is propperty type?

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  private shouldDisplayContactPageToaster(): boolean {
+    const [_root, _page, pageType] = this.router.url.split('/');
+
+    // TODO: change isLoggedIn and isProfileFilled with real services
+    const isLoggedIn = this.toasterService.isLoggedIn();
+    const isProfileFilled = this.toasterService.isProfileFilled();
+
+    return (!isLoggedIn || (isLoggedIn && !isProfileFilled)) && pageType === this.routerNames.Contact;
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-toaster/epm-toaster.component.ts:69]: can we separate it (!isLoggedIn || (isLoggedIn && !isProfileFilled))  to indpendent private method?

File: src/app/services/toaster.service.ts (hunk 1 above)
Comment (HellraiserCW) [reply src/app/services/toaster.service.ts:12]: fixed

File: src/app/shared/components/epm-toaster/epm-toaster.component.html (hunk 2 above)
Comment (HellraiserCW) [reply src/app/shared/components/epm-toaster/epm-toaster.component.html:9]: fixed

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts (hunk 3 above)
Comment (HellraiserCW) [reply src/app/shared/components/epm-toaster/epm-toaster.component.ts:19]: You are right, this is not acceptable variable name for boolean type. Fixed.

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts (hunk 4 above)
Comment (HellraiserCW) [reply src/app/shared/components/epm-toaster/epm-toaster.component.ts:24]: fixed

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts (hunk 5 above)
Comment (HellraiserCW) [reply src/app/shared/components/epm-toaster/epm-toaster.component.ts:69]: Thanks, good idea to separate logic according user state. Fixed.

File: src/app/app.config.ts [hunk 6]
```diff
@@ -31,3 +31,8 @@ export enum BUTTON_THEMES {
   rounded = 'rounded',
   roundedInverted = 'rounded_inverted'
 }
+
+export enum TOASTER_ICONS {
+  Close = 'assets/icons/toaster-icons/close.svg',
+  Info = 'assets/icons/toaster-icons/info.svg'
+}
```
Comment (GennadiyShpak) [inline src/app/app.config.ts:38]: same problem as in previos PR

File: src/app/services/toaster.service.ts (hunk 1 above)
Comment (GennadiyShpak) [reply src/app/services/toaster.service.ts:16]: what do you fix? in your code private and public propertyes are mixed

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+@Component({
+  selector: 'epm-toaster',
+  standalone: true,
+  imports: [CommonModule, NgOptimizedImage, RouterLink],
+  templateUrl: './epm-toaster.component.html',
+  styleUrls: ['./epm-toaster.component.scss']
+})
+export class EpmToasterComponent implements OnInit {
+  @Input() title = '';
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-toaster/epm-toaster.component.ts:17]: Do we need it?

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+  private setupNavigationEventListener(): void {
+    this.router.events
+      .pipe(
+        filter(event => event instanceof NavigationEnd),
+        tap(() => {
+          this.manageContactPageToaster();
+        })
+      )
+      .subscribe();
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-toaster/epm-toaster.component.ts:50]:
```suggestion
      ) .subscribe();
```

File: src/app/services/toaster.service.ts (hunk 1 above)
Comment (HellraiserCW) [reply src/app/services/toaster.service.ts:16]: I can do it only this way. But still private will remain higher.
```suggestion
  private isHidden$: WritableSignal<boolean> = signal(true);
  private toasterSettings$: WritableSignal<ToasterMessage> = signal(initialToasterConfig);
  
  isHidden: Signal<boolean> = computed(this.isHidden$);
  toasterSettings: Signal<ToasterMessage> = computed(this.toasterSettings$);
```

File: src/app/app.config.ts (hunk 6 above)
Comment (HellraiserCW) [reply src/app/app.config.ts:38]: Fixed moving url path to template string literal

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts (hunk 8 above)
Comment (HellraiserCW) [reply src/app/shared/components/epm-toaster/epm-toaster.component.ts:50]: fixed

File: src/app/shared/components/epm-toaster/epm-toaster.component.ts (hunk 7 above)
Comment (HellraiserCW) [reply src/app/shared/components/epm-toaster/epm-toaster.component.ts:17]: fixed

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**